components/threading/virtualized_thread_viewer/reply/reply.tsx

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo, useCallback} from 'react';
+import {useDispatch} from 'react-redux';
+
+import {removePost} from 'mattermost-redux/actions/posts';
+
+import {Post} from 'mattermost-redux/types/posts';
+
+import RhsComment from 'components/rhs_comment';
+
+export type OwnProps = {
```

Comment:
> Is OwnProps used in any other place?

In index.ts
>  In general, for the index files you just need to declare in OwnProps the props you are using for index. In this case it would be `type OwnProps = {id: string;}`. And since we usually only use them in the index file, we define that in the index file.

it seemed a bit easier to reason as to which props are passed from the parent and which are from redux or HOC etc... Will change to conform to our code style :)

> Also, the "main component" usually don't use the props in OwnProps (they are for the index to do extra stuff) so are usually omitted from the props.

@larkox In the case of `Reply` and RootPost` I get the `post` from redux but the other props are passed from the parent, since they are the same for all replies. This was done in an attempt to reduce re-rendering of the list on edited posts.

Do you feel that we should get the other props from index as well?

Reply:
If the state changes, the component will be re-rendered either way, no matter if the change comes from the parent props or the index props.
If the parent is not using the props, better to fetch them on index (that way we don't re-render the parent). If it is using them, I would say there is not much difference. For "clarity" I would pass the props from the parent, but I would say is a style choice.

Reply:
Yep, but if the parent holds a reference to all posts, and one post changes, then all children will get re-rendered.
whereas in the other case we can optimize via `memo` and only the child that has the changed post will get re-rendered.

Reply:
hm, on the case that a post message is edited, and the parent holds all posts, then the parent will get re-rendered,
but If each child holds its post data only one child will get re-rendered.